File: src/main/java/niklasu/speedtester/influx/InfluxConnector.java
package niklasu.speedtester.influx;

import com.influxdb.client.WriteApiBlocking;
import com.influxdb.client.domain.WritePrecision;
import com.influxdb.client.write.Point;
import niklasu.speedtester.config.ConfigProvider;
import niklasu.speedtester.influx.InfluxModule.InfluxWriter;

import java.net.InetAddress;
import java.net.UnknownHostException;
import java.time.Instant;

class InfluxConnector implements InfluxWriter {
    private final ConfigProvider configProvider;
    private final WriteApiBlocking api;

    public InfluxConnector(WriteApiBlocking api, ConfigProvider configProvider) {
        this.configProvider = configProvider;
        this.api = api;
    }

    @Override
    public void write(final double speed) throws UnknownHostException {
        Point point1 = Point.measurement("speed")
                .time(Instant.now(), WritePrecision.MS)
                .addField("value", speed)
                .addField("download_size", configProvider.size())
                .addTag("hostname", InetAddress.getLocalHost().getHostName())
                .addTag("download_url", configProvider.url());
        api.writePoint(point1);
    }
}

File: src/main/java/niklasu/speedtester/influx/InfluxConnector.kt
package niklasu.speedtester.influx

import niklasu.speedtester.config.ConfigProvider
import niklasu.speedtester.influx.InfluxModule.InfluxWriter
import org.influxdb.InfluxDB
import org.influxdb.dto.Point
import java.net.InetAddress
import java.util.concurrent.TimeUnit

class InfluxConnector(private val influx: InfluxDB, private val configProvider: ConfigProvider) : InfluxWriter {
    override fun write(speed: Double) {
        val point1: Point = Point.measurement("speed")
                .time(System.currentTimeMillis(), TimeUnit.MILLISECONDS)
                .addField("value", speed)
                .addField("download_size", configProvider.size)
                .tag("hostname", InetAddress.getLocalHost().hostName)
                .tag("download_url", configProvider.url)
                .build()
        influx.write(point1)
    }
}